text/0000-lwc-signals.md

```diff
@@ -0,0 +1,372 @@
+
+* Identify which properties need to be reactive and create `Signals` for them.
+* Refactor templates to use the `.value` property of Signals where tracked properties were previously used. This guarantees that the child component will receive the value, not the Signal itself.
+* Update component logic to explicitly set the new value of a Signal when the state changes. This can be done by using the `.value` property of the Signal, or by replacing the entire Signal with a new one, which is slightly different from the current `@track` behavior, where you can change a property of the object and the component will react to it.
+
+## Transition from `@wire`
+
+### Background
+
+The `@wire` decorator allows for a declarative approach to data fetching and reactive property updates based on changing inputs. It is closely tied to the Salesforce data layer and service layer.
+
+### New Approach
+
+Under the new design, external stores or Signals would take over the responsibility of managing reactive data fetching and updates. These stores would follow a standard protocol, making them framework-agnostic.
+
+__Advantages__:
+
+* __Flexibility__: Developers can use any data store that adheres to the protocol, not just Salesforce-specific services.
+* __Consistency__: A consistent pattern for managing both local and external data simplifies the mental model for developers.
+
+### Migration Path
+
+#### Replacing a `@wire` property
+
+We can go from this:
+
+```javascript
+import { RecordWireAdapter } from "c/wireAdapters";
+
+export default class ExampleComponent extends LightningElement {
+    @api recordId;
+    @wire(RecordWireAdapter, { recordId: "$recordId" }) record;
+}
+```
+
+To this:
+
+```javascript
+import { RecordStore } from "c/stores";
+
+export default class ExampleComponent extends LightningElement {
+    @api recordId;
+    record = new RecordStore(this.$api.recordId);
+}
+```
+
+This seems simple enough. No major differences, the `$` continues to be a thing in the new model so the store can access the value of the public property and react to it by fetching new data, it is just a different placement for the `$`. In terms of the usage of the record iself, it is almost the same, except that now we need to use `.value` to access the actual record in the template, e.g.:
+
+```html
+<template>
+    <p>{record.value.Name}</p>
+</template>
+```
+
+#### Replacing a `@wire` method
+
+We can go from this:
+
+```javascript
+import { RecordWireAdapter } from "c/wireAdapters";
+
+export default class ExampleComponent extends LightningElement {
+    @api recordId;
+    @wire(RecordWireAdapter, { recordId: "$recordId" }) updateRecord(record) {
+      // do something with the record
+    };
+}
+```
+
+To this:
+
+```javascript
+import { RecordStore } from "c/stores";
+
+export default class ExampleComponent extends LightningElement {
+    @api recordId;
+    record = new RecordStore(this.$api.recordId);
+
+    updateRecord(record) {
```

Comment:
It would be great to flesh out the error handling some more.

- Can creating a store cause an error?
- Do methods have an error object or is there a different mechanism for communicating errors to components?  Error handling frequently breaks static analysis, so if there's a way to handle them separately from the data, that would be nice.

For low-code, we are moving towards completely presentational components that do not make data calls.  The view metadata binds the data to the components, so the framework is responsible for handling errors.  You can envision an admin saying "Show this error component if a data failure occurs".